src/itr_evaluator.ml

```diff
@@ -955,7 +955,20 @@ and evaluate_expr (context : 'a context) (e : expr) : record_item =
     | Rec_value lhs, Rec_value rhs -> check_cmp lhs rhs op
     | _ -> Rec_value e)
   | Value (Funcall { func : value; args : record_item list })
-    when (func = Literal (String "LString.length")||func = Literal (String "String.length")||func = Literal (String "len")) && List.length args = 1 ->
+    when (func = Literal (String "len")) && List.length args = 1 ->
```

Comment:
Perhaps this is unexpected, but the `"len"` function is the translation of both `String.length` and `List.length` [here](https://github.com/imandra-ai/ipl-worker/blob/7e39eedb27218895d337a04cd78a4b27b2299893/imandra-test-runner/region_to_itr.ml#L132-L143). We may to fix the transpilation as well (but this change is probably helpful for existing files in any case)

Reply:
Yes this looks good - some of these were to deal with legacy cases. The "len" was introduced so the constraints looks nicer printed but that could be moved into the printer.

Reply:
Ah nice, yeah I'll have some function name prettifiers in the UI anyway. Could have something similar in the pretty printers but in some ways it's nice to go straight from the raw pp text to the relevant evaluator case without having a layer of indirection